aptos-move/framework/src/module_metadata.rs

```diff
@@ -228,6 +228,8 @@ pub fn get_metadata_v0(md: &[Metadata]) -> Option<Arc<RuntimeModuleMetadataV1>>
 
 /// Extract metadata from the VM, upgrading V0 to V1 representation as needed
 pub fn get_vm_metadata(vm: &MoveVM, module_id: &ModuleId) -> Option<Arc<RuntimeModuleMetadataV1>> {
```

Comment:
But they do dispatch different functions? `get_metadata_v0` and `get_metadata`? 

Right now we have 
```
thread_local! {
    static V1_METADATA_CACHE: RefCell<LruCache<Vec<u8>, Option<Arc<RuntimeModuleMetadataV1>>>> = RefCell::new(LruCache::new(METADATA_CACHE_SIZE));

    static V0_METADATA_CACHE: RefCell<LruCache<Vec<u8>, Option<Arc<RuntimeModuleMetadataV1>>>> = RefCell::new(LruCache::new(METADATA_CACHE_SIZE));
}
```
and my plan (let's see if this will work haha) is to have `AptosModuleStorage: ModuleStorage` which will allow you to fetch metadata, which is cached in code cache as well. So no need for these caches, because you can just call `fetch_module_aptos_metadata<M>(address, module_name) -> PartialVMResult<&M>`. But current metadata implementation is a dark forest and was avoiding it so far🙃

Reply:
oh didn't see that part..